cheatsheets/Authentication_Cheat_Sheet.md

```diff
@@ -25,12 +25,9 @@ A key concern when using passwords for authentication is password strength. A "s
 
 - Password Length
     - **Minimum** length of the passwords should be **enforced** by the application. Passwords **shorter than 8 characters** are considered to be weak ([NIST SP800-63B](https://pages.nist.gov/800-63-3/sp800-63b.html)).
-    - **Maximum** password length should not be set **too low**, as it will prevent users from creating passphrases. Typical maximum length is 128 characters. It is important to set a maximum password length to prevent [long password Denial of Service attacks](https://www.acunetix.com/vulnerabilities/web/long-password-denial-of-service/).
-    - Some password hashes such as Bcrypt truncate the input, so a shorter maximum length may be required, as discussed in the [Password Storage Cheat Sheet](Password_Storage_Cheat_Sheet.md#maximum-password-lengths).
+    - **Maximum** password length should not be set **too low**, as it will prevent users from creating passphrases. A common maximum length is 64 characters due to limitations in certain hashing algorithms, as discussed in the [Password Storage Cheat Sheet](Password_Storage_Cheat_Sheet.html#maximum-password-lengths). It is important to set a maximum password length to prevent [long password Denial of Service attacks](https://www.acunetix.com/vulnerabilities/web/long-password-denial-of-service/).
 
-      When selecting a maximum password length, consider whether the hashing algorithm to be used has any limitations because some [have a maximum password length](https://security.stackexchange.com/questions/39849/does-bcrypt-have-a-maximum-password-length/39851#39851).
-
-- Do not truncate passwords. Make sure that every character the user types in is actually included in the password.
+- Do not silently truncate passwords. If the user enters a password that is longer than that maximum permitted length then return an error message explaining to them what the problem is.
```

Comment:
Don't you think pre-hashing could be referenced in this point from pass storage CS?
Maybe create 2 bullets under `Do not silently truncate password` one for each option.

Reply:
The sections linked the the bullet point above, so it seems a little unnecessary to link again here. Even if you're pre-hashing you still need to have some kind maximum length (although probably a very high one), so I think the point still stands.

Reply:
Then you should be removing everything after `Do not silently truncate passwords`, as the rest is part of the maximum length. I am just following the logic of what's written.

Reply:
I think that telling people to present an error to the user here is valid - there's no conflict between that and the previous point. @mackowski do you have a strong feeling either way?

Reply:
The idea is the following:
In the Maximum length recommendation, it is instructed to either block the user from setting a length longer than the max allowed, or use some kind of a pre-hash.
In the Do Not Truncate, which happens if the user goes above the max length, they have the same 2 options ...
A cleaner thought could be to say:
> Do not truncate passwords. Refer to the maximum length measures to be taken.
Something similar to that.

I am not saying that presenting an error is not valid!

Reply:
I think that simply:
`Do not silently truncate passwords. See details in [Password Storage Cheat Sheet](Password_Storage_Cheat_Sheet.html#maximum-password-lengths) how to handle scenario when user enters a password that is longer than that maximum permitted length`
I think that there is nothing bad in linking here again to that CS. In that place it is well described so anything we put here will be only a simplification.

Reply:
Implement a slight variation on the wording that @mackowski suggested.